compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -21,6 +22,75 @@ use serde::{Deserialize, Serialize};
 use serde_with::serde_as;
 use serde_with::DisplayFromStr;
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub(crate) struct CallStackId(u32);
+
+impl CallStackId {
+    pub(crate) fn root() -> Self {
+        Self::new(0)
```

Comment:
Can you move call stack related things to their own file?

Also, since we no longer need the linked list type - can you remove that file too?